gms/gossiper.cc

```diff
@@ -1814,8 +1814,12 @@ future<> gossiper::start_gossiping(int generation_nbr, std::map<application_stat
     _enabled = true;
     _nr_run = 0;
     _scheduled_gossip_task.arm(INTERVAL);
+    _background_msg = seastar::gate();
```

Comment:
Instead of dropping the old _background_msg directly,  it is better to close it first. 

```
--- a/gms/gossiper.cc
+++ b/gms/gossiper.cc
@@ -1823,8 +1823,14 @@ future<> gossiper::start_gossiping(int generation_nbr, std::map<application_stat
     _enabled = true;
     _nr_run = 0;
     _scheduled_gossip_task.arm(INTERVAL);
+    if (!_background_msg.is_closed()) {
+        co_await _background_msg.close();
+    }
+    _background_msg = seastar::gate();
```

Reply:
Done